sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -307,8 +310,8 @@ case class GetMapValue(child: Expression, key: Expression)
           }
         }
 
-        if ($found) {
-          ${ev.value} = ${ctx.getValue(eval1 + ".valueArray()", dataType, index)};
+        if ($found && !$values.isNullAt($index)) {
```

Comment:
nit: to be consistent with the interpreted version, we can switch the `if` and `else` branches